llvm/lib/Target/AMDGPU/AMDGPUMachineFunction.cpp

```diff
@@ -102,6 +103,12 @@ unsigned AMDGPUMachineFunction::allocateLDSGlobal(const DataLayout &DL,
 
   unsigned Offset;
   if (GV.getAddressSpace() == AMDGPUAS::LOCAL_ADDRESS) {
+    if (TargetExtType *TTy = AMDGPU::isNamedBarrier(GV)) {
+      auto Offset = getLDSAbsoluteAddress(GV);
+      assert(Offset && "named barrier should have an absolute address");
```

Comment:
This doesn't seem like something appropriate to assert on. Is there a verifier check for this?

Reply:
Changed it.